scripts/split-mems-conf.py

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env python
+
+import os
+import json
+import argparse
+from typing import List, Optional
+
+# Schema of json emitted by circt
+"""
+{
+  "module_name": "mem_ext",
+  "depth": 512,
+  "width": 64,
+  "masked": true,
+  "read": false,
+  "write": false,
+  "readwrite": true,
+  "mask_granularity": 8,
+  "extra_ports": [],
+  "hierarchy": [
+    "TestHarness.ram.srams.mem.mem_ext"
+  ]
+}
+"""
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description='Use CIRCT (firtool) smems JSONs to create DUT and test harness smems confs')
+    parser.add_argument('--in-smems-conf', type=str, required=True, help='')
+    parser.add_argument('--in-dut-smems-json', type=str, required=True, help='')
+    parser.add_argument('--in-tb-smems-json', type=str, required=True, help='')
+    parser.add_argument('--out-dut-smems-conf', type=str, required=True, help='')
+    parser.add_argument('--out-tb-smems-conf', type=str, required=True, help='')
+    args = parser.parse_args()
+
+    with open(args.in_smems_conf) as isc:
+        with open(args.in_dut_smems_json) as idsj:
```

Comment:
Try doing the multline thing when opening multiple files at once

https://legacy.python.org/dev/peps/pep-0008/#maximum-line-length

Reply:
Done